In [1]:
import pandas as pd

### No ML strategies

In [8]:
df = pd.read_csv("../data/processed/strategy_1_5_results.csv")

df

,strategy,nb_bets,prop_winning_bets,final_gain,max_cumulative_gain,max_cumulative_loss,max_gain_day,max_loss_day,roi,strat_score
0,strategy_1,602,0.531561,38.0,53.0,-1.0,4.0,-4.0,0.063123,7.043189
1,strategy_2_1,255,0.666667,31.0,33.0,-3.0,3.0,-3.0,0.121569,7.333333
2,strategy_2_2,477,0.605870,11.0,26.0,1.0,4.0,-4.0,0.023061,3.938155
3,strategy_3,602,0.531561,313.0,320.0,-33.0,129.0,-96.0,0.127599,1.771872
4,strategy_4,602,0.531561,305.0,310.0,-21.0,255.0,-128.0,0.148274,1.287375
5,strategy_5_1,255,0.666667,113.0,114.0,-22.0,63.0,-32.0,0.146184,2.375000
6,strategy_5_2,477,0.605870,155.0,182.0,1.0,64.0,-32.0,0.119139,3.445886
7,strategy_5_3,581,0.621343,216.0,229.0,-39.0,64.0,-34.0,0.130988,4.184925


In [9]:
print(df[df["strat_score"]==df["strat_score"].max()])

       strategy  nb_bets  prop_winning_bets  final_gain  max_cumulative_gain  \
1  strategy_2_1      255           0.666667        31.0                 33.0   

   max_cumulative_loss  max_gain_day  max_loss_day       roi  strat_score  
1                 -3.0           3.0          -3.0  0.121569     7.333333  


In [10]:
df_sorted = df.sort_values(by="strat_score", ascending=False)
df_sorted

,strategy,nb_bets,prop_winning_bets,final_gain,max_cumulative_gain,max_cumulative_loss,max_gain_day,max_loss_day,roi,strat_score
1,strategy_2_1,255,0.666667,31.0,33.0,-3.0,3.0,-3.0,0.121569,7.333333
0,strategy_1,602,0.531561,38.0,53.0,-1.0,4.0,-4.0,0.063123,7.043189
7,strategy_5_3,581,0.621343,216.0,229.0,-39.0,64.0,-34.0,0.130988,4.184925
2,strategy_2_2,477,0.605870,11.0,26.0,1.0,4.0,-4.0,0.023061,3.938155
6,strategy_5_2,477,0.605870,155.0,182.0,1.0,64.0,-32.0,0.119139,3.445886
5,strategy_5_1,255,0.666667,113.0,114.0,-22.0,63.0,-32.0,0.146184,2.375000
3,strategy_3,602,0.531561,313.0,320.0,-33.0,129.0,-96.0,0.127599,1.771872
4,strategy_4,602,0.531561,305.0,310.0,-21.0,255.0,-128.0,0.148274,1.287375


### ML strategies

In [11]:
df = pd.read_csv("../data/processed/strategy_6_results.csv")

df

,strategy,nb_bets,prop_winning_bets,final_gain,max_cumulative_gain,max_cumulative_loss,max_consecutive_losses,nb_three_or_more_consecutive_losses,max_gain_day,max_loss_day,roi,strat_score
0,strategy_6_noML,108,0.472222,44,51,-89,7,23,128,-96,0.072368,3.440476
1,strategy_6_lr_0.5,76,0.513158,38,39,-1,4,20,16,-8,0.191919,5.003289
2,strategy_6_lr_0.53,56,0.535714,29,30,-40,6,30,64,-32,0.111969,2.678571
3,strategy_6_lr_0.55,40,0.650000,25,26,0,3,9,8,-4,0.384615,5.633333
4,strategy_6_lr_0.6,18,0.722222,13,13,0,1,0,2,-1,0.565217,9.388889
5,strategy_6_lr_0.65,7,0.857143,6,6,0,1,0,2,-1,0.750000,5.142857
6,strategy_6_xgb_0.5,58,0.465517,24,27,-111,7,30,128,-64,0.064516,1.795567
7,strategy_6_xgb_0.53,53,0.452830,21,24,-113,7,33,128,-64,0.046771,1.552561
8,strategy_6_xgb_0.55,51,0.470588,21,24,-62,6,27,64,-32,0.065831,1.882353
9,strategy_6_xgb_0.6,42,0.476190,19,20,-53,6,35,64,-32,0.085202,1.587302


In [12]:
print(df[df["strat_score"]==df["strat_score"].max()])

            strategy  nb_bets  prop_winning_bets  final_gain  \
4  strategy_6_lr_0.6       18           0.722222          13   

   max_cumulative_gain  max_cumulative_loss  max_consecutive_losses  \
4                   13                    0                       1   

   nb_three_or_more_consecutive_losses  max_gain_day  max_loss_day       roi  \
4                                    0             2            -1  0.565217   

   strat_score  
4     9.388889  


In [13]:
df_sorted = df.sort_values(by="strat_score", ascending=False)
df_sorted

,strategy,nb_bets,prop_winning_bets,final_gain,max_cumulative_gain,max_cumulative_loss,max_consecutive_losses,nb_three_or_more_consecutive_losses,max_gain_day,max_loss_day,roi,strat_score
4,strategy_6_lr_0.6,18,0.722222,13,13,0,1,0,2,-1,0.565217,9.388889
3,strategy_6_lr_0.55,40,0.650000,25,26,0,3,9,8,-4,0.384615,5.633333
5,strategy_6_lr_0.65,7,0.857143,6,6,0,1,0,2,-1,0.750000,5.142857
1,strategy_6_lr_0.5,76,0.513158,38,39,-1,4,20,16,-8,0.191919,5.003289
0,strategy_6_noML,108,0.472222,44,51,-89,7,23,128,-96,0.072368,3.440476
2,strategy_6_lr_0.53,56,0.535714,29,30,-40,6,30,64,-32,0.111969,2.678571
10,strategy_6_xgb_0.65,32,0.500000,15,16,-7,4,20,16,-8,0.172414,2.000000
8,strategy_6_xgb_0.55,51,0.470588,21,24,-62,6,27,64,-32,0.065831,1.882353
6,strategy_6_xgb_0.5,58,0.465517,24,27,-111,7,30,128,-64,0.064516,1.795567
9,strategy_6_xgb_0.6,42,0.476190,19,20,-53,6,35,64,-32,0.085202,1.587302


Logistic regression gives better results than xgboost, regardless of the threshold used.
For minimal risk, lr with a threshold between 0.6 and 0.65 is a good choice.
A threshold of 0.55 also seems interesting because the maximum number of consecutive losses does not exceed 3.